### ISSUE TITLE
It's Working On Xarmain - Android ?

### ISSUE THREAD

**Issue by GeeSuth** [gpailler/MegaApiClient#79]
hey! before anything , it's amazing library ..
i was use it on C# desktop application and good working .. 

but i recently i need to work with android app.. 

I'm try to use async method .. but don't good working 
and the bad in this the error not known not handle "An unhandled exception occurred"
because this i don't know what the exception exactly !!!!!

i'm wrote this :   
`     private async Task<string> UploadImage(Stream file)
        {
            MegaApiClient Newclient = new MegaApiClient();
            await Newclient.LoginAsync("Email", "Password");
            var nodes = Newclient.GetNodes();
            INode root = nodes.Single(n => n.Type == NodeType.Root);
            System.IProgress<double> progress = new System.Progress<double>();
            INode MyFile = await Newclient.UploadAsync(file, "toto.jpg", root, progress);
            System.Uri dawnloadurl = await Newclient.GetDownloadLinkAsync(MyFile);
            return dawnloadurl.ToString();
        }`

when i try breakpoint  and step by step with code 
i note the problem with **LoginAsync** & **Login** !!

Note: When i try on Desktop application it's prefect working 

Thank you Mr.gpailler

**Comment by gpailler** [comment]
I don't know if this library is working on Xamarin. I never tried.
Could you test with the non async version `NewClient.Login("User", "Password")`?
About the exception, do you have any exception prior this one in the Console (or maybe an inner exception) ?

**Comment by GeeSuth** [comment]
I can't Handle any error !! 
this only show "An unhandled exception occurred" 
and no any details ..
you should try it on Xarmain !! 
i really need it ...
but i'm sure the problem on Login & LoginAsync ...  
because i'm check the IsLogin properties it's false !! 

Please Help our Sir!!
I will waiting for your feedback 
Thank you

**Comment by gpailler** [comment]
Could you send me a minimal project to reproduce the issue?
(You can attach a zip file to this issue)

**Comment by GeeSuth** [comment]
Sure. Why not!  Give me some hour because I'm work in my jop .. 
I'm will send it .. 
Thank you !

**Comment by GeeSuth** [comment]
Sorry for Delay anywhere 
I'm try to upload the ZIP file but github don't allow More than 10MB

I'm Uploaded it on Mega .. 
this link if you dont mind 

> https://mega.nz/#!gipFhApJ!7Sph0G_utVaZMiWCLm1m2kM0fykB8eKKT5HjgECI51A

I'm confusing if the MegaApiClient supported Mono? 

Thank you bro .. I'll waiting for you!!

**Comment by gpailler** [comment]
Thanks for the project. I found two issues with MegaApiClient library on Android.

- The Xamarin implementation of Aes doesn't support transform reuse (it was an optimization in MegaApiClient). I will fix it.
- Because of Xamarin linking optimization, some properties are removed from the code and the Json deserialization doesn't work as expected. The solution is to add `MegaApiClient` in the **Skip linking assemblies** section.
![image](https://user-images.githubusercontent.com/3621529/32175930-e62df230-bdb8-11e7-89e1-9584cb931bf0.png)

**Comment by GeeSuth** [comment]
No problem .. i'm sure you doing well .. 
i need to know when you'll finished fixed ..

Can you send notification  to me ? or

**Comment by gpailler** [comment]
Could you test latest pre-release build available on MyGet (https://www.myget.org/feed/megaapiclient/package/nuget/MegaApiClient/1.6.2-master-376)?

The requested changes in your code are:
- Add MegaApiClient in skip linking assemblies
- Change the following in your sample
```csharp
file.Position = 0; // Reset position to 0 
System.IProgress<double> progress = new System.Progress<double>();
INode MyFile = await Newclient.UploadAsync(file, "toto.png", root, progress); // Extension is PNG, not JPG
```

**Comment by gpailler** [comment]
Fixed in 1.6.2

**Comment by GeeSuth** [comment]
Sorry!
Yeeeeeeeeeah ! it's Amazing fix! 
it's working now!
